components/ca-form-builder.css

```diff
@@ -0,0 +1,221 @@
+body {
+    background-color: #eee;
+}
+ca-form-builder {
+    position: relative;
+    top: 10px;
+    min-height: 400px;
+    min-width: 600px;
+    display: block;
+}
+ca-form-builder ca-form {
+    overflow: auto;
+    display: block;
+}
+/* place an outline around the selected element */
+
+ca-form-builder label[data-selected="true"] {
+    outline: -webkit-focus-ring-color auto 5px;
+    border: 1px solid #666666;
+}
+ca-form-builder label[data-selected="true"] input,
+ca-form-builder label[data-selected="true"] select,
+ca-form-builder label[data-selected="true"] textarea {
+    outline: none;
+}
+/* remove the outline around the selected label when the form builder is in "text-edit" mode */
+
+ca-form-builder[data-mode="text-edit"] label[data-selected="true"] {
+    outline: none;
+}
+/* fix the width of the time element of date-time controls */
+
+ca-form-builder ca-form label[data-format="date-time"] ca-datetime input[type="time"],
+ca-form-builder ca-form label[data-format="local-date-time"] ca-datetime input[type="time"] {
+    width: 90px;
+}
+.ca-form-builder-preview form[data-selected="true"] {
+    outline: -webkit-focus-ring-color auto 5px;
+    border: 1px solid #666666;
+}
+ca-form-builder .ca-form-builder-preview ca-form label:after,
+ca-form-builder .ca-form-builder-preview ca-form label.required:after {
+    content: attr(data-key);
+    color: #333333;
+    font-size: 10px;
+    -webkit-transition: all .25s;
+    transition: all .25s;
+    position: absolute;
+    right: 2px;
+    top: 12px;
+    background-color: #ffee00;
+    padding: 0 4px;
+}
+ca-form-builder .ca-form-builder-preview ca-form label[data-selected="true"]:after,
+ca-form-builder .ca-form-builder-preview ca-form label.required[data-selected="true"]:after {
+    color: red;
+    font-size: 14px;
+    top: 8px;
+    -webkit-transition: all .25s;
+    transition: all .25s;
+}
+/* style the container for the toolbox buttons */
```

Comment:
I don't mind comments when they explain something I can't ascertain for myself in 0.3 seconds!